### PR TITLE
Add documentation for ActiveSupport::StringInquirer [ci skip]

### PR DESCRIPTION
**Pull request by kyatul**

### CODE REVIEW

**Comment by kaspth** [comment]
Thanks, but I don't think exposing implementation details improves the documentation here. I find the old docs just as clear 😁

**Comment by kyatul** [comment]
Previously it just consisted of one use case of **Rails.env**, saying nothing about how to instantiate a new object of StringInquirer. Adding an example describing how to instantiate one will make things more clear on first impression.

File: activesupport/lib/active_support/string_inquirer.rb [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+  # The value returned by <tt>Rails.env</tt> is wrapped
   # in a StringInquirer object, so instead of calling this:
   #
-  #   Rails.env == 'production'
+  # Rails.env == 'production'
   #
   # you can call this:
   #
-  #   Rails.env.production?
+  # Rails.env.production?
```
Comment (kaspth) [inline activesupport/lib/active_support/string_inquirer.rb]: Let's add any how to instantiate this documentation below here. Also the two space indentation on code wasn't a fluke, it's on purpose. Please check out our documentation guidelines: http://guides.rubyonrails.org/api_documentation_guidelines.html 😁

**Comment by kaspth** [comment]
@kyatul alright, I'll bite. Let's append to the existing documentation instead of inserting into it. I think it has a nice flow and clearly states the motivation. Though I do agree an instantiation example would be nice.

File: activesupport/lib/active_support/string_inquirer.rb (hunk 1 above)
Comment (kyatul) [inline activesupport/lib/active_support/string_inquirer.rb]: Yeah, I missed on that spaces thing. My bad!

**Comment by kyatul** [comment]
@kaspth Thanks for finally considering it. I have raised a new PR #26036 ,after making your above said changes :)

**Pull request closed without merging**